### DISCUSSION TITLE
Retrieving Customer Username and Profile Picture via Etsy API v3

### DISCUSSION THREAD

**Discussion by jinspe** [etsy/open-api#1070] [Q&A]
Hey Etsy Community,

I hope you're all doing well. I've been grappling with an issue while trying to fetch customer **usernames** and **profile pictures** using Etsy **API v3**. 

In the previous version (v2), it was straightforward with the "/users/:user_id/profile" and "/users/:user_id/avatar/src" endpoints. But now, in v3, I seem to be hitting a dead end.

Any ideas on how to address this? Your help would be greatly appreciated!
Thank you

**Comment by Gareth064** [comment]
https://developer.etsy.com/documentation/reference#operation/getUser

This endpoint.

You give it the buyerid from the receipt.

But you don't get her their username anymore, so you can't actually do anything to find their profile on Etsy.

If the buyer checked out as a guest , or in many cases using their apple account, then you dont get a result from this API, so don't expect it to work all the time.

**Reply by bizmate** [reply]
the user resource in V3 does not work without an oauth token compared to the v2 where all these resources were open and accessible with the APIkey only. Can this be opened so that an oauth is not needed if all we are looking up are public user information .

This is the error i hit in V2 and there is no alternative in V3 because this resource is locked behind a Oauth requirement.

An extract from the logs of our app now completely broken

```
```EtsyClient::getUserProfile for user id: 493864934 threw ClientException with message: Client error: `GET https://openapi.etsy.com/v2/users/493864934/profile?api_key=XXXXX` resulted in a `410 Gone` response:
{"error":"API v2 deprecated.",  "message":"The API version 2 you are trying to access has been deprecated. 
```

**Reply by Gareth064** [reply]
There is no equivalent endpoint in v3, OAuth or not. You can no longer get Etsy user details via their API. Only shop owners see the "buyer" user details form an Order. And even then, that might be protected by an Apple user, it doesn't even give their name or email address.

**Reply by bizmate** [reply]
Users are not just part of Orders. They are returned in the so called getReviewsByShop such as ... "buyer_user_id": 15943556
The lack of such details means one is supposed to scrape it ?
I had raised this suggestion over a year ago, I think under https://github.com/etsy/open-api/issues/272 or https://github.com/etsy/open-api/issues/470 as per my notes but Etsy changed the structure of this project on github so the issues structure is gone.

Should businesses interested in reviews really do scraping? By the way this is also a problem for the amount of API calls allowed. It takes a lot of requests to fetch reviews. I am not sure if you are part of the Etsy team but hopefully this gets changed so that other businesses like mine in data integration can advertise Etsy for free.

**Reply by Gareth064** [reply]
I'm not part of Etsy no, but I've been in this repo since Etsy started building v3. I don't remember everything that's been discussed in the past couple of years, and annoyingly Etsy deleted the issues section so we don't have those to look back on.... But from vague memory, Etsy stopped the ability to get user details, using the buyer Id, due to privacy reasons.

My app doesn't deal with reviews so I can't relate to what you're trying to do, but why do you need the users details who left the review? Isn't the review contents itself not enough?

**Reply by bizmate** [reply]
In every legit plaform out there you can see the reviews users, Google, Facebook etc but also it is rather inconsistent that you can see this information on the actual Etsy ``public`  site but not through the API. If you see all reviews show who left them so why not show this info in the API?

**Reply by Gareth064** [reply]
Yeah I get what you mean.

I think I mentioned this recently on a different discussion, we can't think of Etsy's API like we would with the Shopify API or similar. Etsy wants buyers and sellers to be logged into their Websites/Apps to interact with them, they don't really have a desire to build an amazing API which can allow interactions with Etsy data outside of their Etsy Websites/Apps. 

Which is a shame, because most of us developers are trying to build apps which plug feature gaps in Etsy, or simple quality of life features which Etsy never seem to want to add to their own platform. Like in my app, it allows orders to be assigned to different users (my app users) so work can be divided out when you have a couple of people working in your small business. Sellers has been asking for years to allow multiple user accounts to log into Etsy Shops to help manage orders, instead of sharing accounts/passwords.... and they just won't do it.

**Reply by bizmate** [reply]
Thank you for sharing your use case. I guess it becomes very complex but if you can do all operations then your app can indeed provide the functionality by giving a single user in the company the GOD powers/super admin. They can set up the account with your app and then your app could potentially dispatch orders (not sure if applicable) for the management. Then you have virtual etsy users in your app instead of being in Etsy and you could potentially build all the Auth mechanism so each of them can perform all or some operations only, never having to use the master password and account.

In my case the only solution is scrapying and it really is bad :(

**Reply by Gareth064** [reply]
> They can set up the account with your app and then your app could potentially dispatch orders (not sure if applicable) for the management. Then you have virtual etsy users in your app instead of being in Etsy and you could potentially build all the Auth mechanism so each of them can perform all or some operations only, never having to use the master password and account.

Yeah this is the point of my app. It allows the shop owner to give access to "just enough" data to allow their staff to create the items. That way the shop owner doesn't give each employee the master username and password to the Etsy shop which would mean they can do what ever they wanted in the Etsy shop, fulfill orders, look at financials, do refunds etc. My app gives levels of security and has auditing to see what each user has done etc. Just your basic multi-user app in reality with different levels of access.

Regarding scraping... I used to tell people not to mention it here in the Etsy repo because scraping is against their terms of service but since Etsy staff rarely make an appearance here anymore, your probably safe there lol

**Reply by bizmate** [reply]
:) I am aware. I do not actually implement the scraping yet and not planning to in the foreseeable future anyway. I have a full solution to do almost everything but I am not integrating Etsy yet just because of this. Scraping is error prone and requires a lot of extra management. Hence I will not publish the Etsy integration yet. My pipeline is very well and busy enough with other work so I will look back into it when possible. It is a shame they actually deprecated V2 or I could I gone live with it. Glad I did not for now

**Reply by Gareth064** [reply]
My app is very specific to helping Etsy Sellers, I wish I could pivot it away, but most other selling platforms have the features my app has already :(

In saying that, I do have integrations with Shopify to sync orders as well, but nothing fancier than that. And that's only because my wife uses Etsy and Shopify. If she didn't use Shopify then I probably wouldn't have integrated it. She's looking at putting items on eBay soon, so will probably integrate it too. But the heart of my app is simple, aggregate orders from multiple Etsy Shops + other selling platforms, then allow staff to have their own queue of work, regardless of where the order came from etc.
